apps/pipelines/nodes/nodes.py

```diff
@@ -127,9 +130,14 @@ def _get_context(self, input, state: PipelineState, prompt: ChatPromptTemplate,
 
         return context
 
-    def _get_history(self, session: ExperimentSession, node_id: str):
+    def _get_history(self, session: ExperimentSession, node_id: str, input):
         if self.history_type == PipelineChatHistoryTypes.GLOBAL:
-            return session.chat.get_langchain_messages_until_summary()
+            return compress_chat_history(
+                chat=session.chat,
+                llm=self.get_chat_model(),
+                max_token_limit=self.max_token_limit,
+                input_messages=[HumanMessage(content=input)],
```

Comment:
This should include the prompt

Reply:
https://github.com/dimagi/open-chat-studio/pull/693/commits/72445d4572b68bbd14168420912257744e846f16